scheduler/src/cook/mesos/util.clj

```diff
@@ -312,22 +313,31 @@
                                    (-> 1 t/hours t/in-millis)))
           expanded-end (Date. (+ (.getTime end)
                                  (-> 1 t/hours t/in-millis)))
-          jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
+          entid-start (d/entid-at db :db.part/user expanded-start)
+          entid-end (d/entid-at db :db.part/user expanded-end)
+          job-user-entid (d/entid db :job/user)]
+          (->> (d/seek-datoms db :avet :job/user user entid-start)
+               (take-while #(and (< (:e %) entid-end)
+                                 (= (:a %) job-user-entid)
+                                 (= (:v %) user)))
+               (map #(:e %))
                (map (partial d/entity db))
                (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
                (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
-               (filter #(= :job.state/completed (:job/state %)))
-               (filter #(not (:job/custom-executor %))))]
-      (->>
-        (cond->> jobs
+               (filter #(= state-keyword (:job/state %)))
+               (filter #(not (:job/custom-executor %)))))))
+
+(defn get-completed-jobs-by-user
```

Comment:
Can we just get rid of this function as well as `get-active-jobs-by-user-and-state` now that they're basically identical and collapse the other logic into `get-jobs-by-user-and-states`?

Reply:
They're not identical as get-completed-jobs-by-user also can check task status.

Reply:
:+1: